websites/C/Coub/presence.ts

```diff
@@ -1,247 +1,370 @@
+                  url: `${document.location.origin}/${
+                    document.location.pathname.split("/")[1]
+                  }`
+                }
+              ]
+            );
+          if (showBridge)
+            data.buttons.push({
+              label: "Download via Bridge",
+              url: parseBridgeUrl(activeMedia.dataset.permalink)
+            });
+          return data;
+        }
+      },
+      {
+        middleware: (ref) => !!ref.location.pathname.match(/^\/view\/(.*)/gi),
+        exec: (
+          context,
+          data,
+          { strings, showBridge, showWatch }: ExecutionArguments
+        ) => {
+          if (!context) return null;
+          const activeMedia = document.querySelector<HTMLElement>(
+            ".coub[coub-block]"
+          );
+          if (!activeMedia) return null;
+          const title = activeMedia
+              .querySelector(".coub__description h5.description__title")
+              ?.textContent?.trim(),
+            isLiked = !!activeMedia.querySelector(
+              ".coub__like-button[widget-like-button].-on"
+            );
 
-    if (!Routes[1] || Routes[1] === "recent") {
-      data.state = "Recent coubs";
-    } else if (Routes[1] === "top") {
-      data.state = "Top coubs";
-    } else if (Routes[1] === "views") {
-      data.state = "Views count coubs";
-    } else if (Routes[1] === "oldest") {
-      data.state = "Oldest coubs";
-    }
-  } else if (Routes[0] === "view") {
-    if (Routes[1]) {
-      data.details = document.querySelector(
-        ".coub__description .description__info h5.description__title"
-      ).textContent;
-      data.state = document.querySelector(
-        ".coub__description .description__info .description__stamp"
-      ).textContent;
-    }
-  } else if (Routes[0] === "friends") {
-    // Friends page - useless page
-    data.details = "Friends";
-  } else if (Routes[0] === "account") {
-    // Account page - wotk in progress
-    data.details = "My account";
-  } else if (Routes[0] === "search") {
-    // Search page
-    data.details = "Search: " + Queries.q.replace(/\W\d+/g, " ");
+          if (!title) return null;
+          data.state = strings.watching;
+          data.details = `${title}${isLiked ? " (❤)" : ""}`;
+          const sourceLink = activeMedia.parentElement.querySelector<HTMLAnchorElement>(
+            `.coub__info .media-block__item > a[type="embedPopup"]`
+          );
+          data.buttons = [];
+          if (showWatch)
+            data.buttons.push(
+              ...[
+                ...getSourceLink(sourceLink?.href),
+                {
+                  label: "Watch on Coub",
+                  url: document.location.href
+                }
+              ]
+            );
+          if (showBridge)
+            data.buttons.push({
+              label: "Download via Bridge",
+              url: parseBridgeUrl(activeMedia.dataset.permalink)
+            });
```

Comment:
here again

Reply:
yes, it will override the last button (only for users that want that implementation)

Reply:
i've added a condition that explicitly replaces the second button (and moved watch coub button to top) for the coub bridge